docs: Update documentation structure with Platform as the first class citizen (#8269)
Significant-Gravitas/AutoGPT: docs/content/classic/index.md

> # AutoGPT Documentation
> 
> Welcome to the AutoGPT Documentation.
> 
> The AutoGPT project consists of four main components:
> 
> - The [Server](#server) &ndash; known as the "AutoGPT Platform"
> - The [Agent](#agent) &ndash; also known as just "AutoGPT"
> - The [Benchmark](#benchmark) &ndash; AKA `agbenchmark`
> - The [Forge](#forge)
> - The [Frontend](#frontend)
> 
> To tie these together, we also have a [CLI] at the root of the project.

Reviewer: Without context, this is confusing.
```suggestion
# AutoGPT Classic Documentation

Welcome to the AutoGPT Classic Documentation.

The classic AutoGPT project consists of four main components:

- The [Agent](#agent) &ndash; also known as just "AutoGPT"
- The [Benchmark](#benchmark) &ndash; AKA `agbenchmark`
- The [Forge](#forge)
- The [Frontend](#frontend)

To tie these together, we also have a [CLI] in `classic/`.
```

## 🌐 Server

<!-- Setup, then Advanced, then New Blocks -->

> **[📖 Setup](server/setup.md)**

Reviewer 2: All of these references need updated since the file moved

&ensp;|&ensp;
**[📖 Advanced Setup](server/advanced_setup.md)**
&ensp;|&ensp;
**[📖 Making New Blocks](server/new_blocks.md)**

> The server is the backbone of the New AutoGPT project. It provides the infrastructure for the agents to run, and the UI for you to interact with them. It integrates with the Forge, Agent, and a bespoke UI to provide a seamless experience.

Reviewer: Not part of the classic project -> shouldn't be in here
